### PR TITLE
Add docs on scheduling functions in new convenience features section

### LINKED ISSUE #912: Add documentation for scheduling
Most of the example work is already done, see 
```
Scheduling Functions
^^^^^^^^^^^^^^^^^^^^

- Added support for scheduling functions more often than the robot loop via ``addPeriodic()`` in TimedRobot. Previously, teams had to make a Notifier to run feedback controllers more often than the TimedRobot loop period of 20ms (running TimedRobot more often than this is not advised). Now, users can run feedback controllers more often than the main robot loop, but synchronously with the TimedRobot periodic functions so there aren't any thread safety issues.

.. tabs::

    .. group-tab:: Java

        .. code-block:: java

            public class Robot {
                private Joystick m_joystick = new Joystick(0);
                private Encoder m_encoder = new Encoder(1, 2);
                private Spark m_motor = new Spark(1);
                private PIDController m_controller = new PIDController(1.0, 0.0, 0.5, 0.01);

                public Robot() {
                    addPeriodic(() -> {
                        m_motor.set(m_controller.calculate(m_encoder.getRate()));
                    }, 0.01, 0.01);
                }

                @Override
                public teleopPeriodic() {
                    if (m_joystick.getRawButtonPressed(1)) {
                        if (m_controller.getSetpoint() == 0.0) {
                            m_controller.setSetpoint(30.0);
                        } else {
                            m_controller.setSetpoint(0.0);
                        }
                    }
                }

    .. group-tab:: C++ (Header)

        .. code-block:: cpp

            class Robot {
                private:
                    frc::Joystick m_joystick{0};
                    frc::Encoder m_encoder{1, 2};
                    frc::Spark m_motor{1};
                    frc2::PIDController m_controller{1.0, 0.0, 0.5, 10_ms};

                    Robot();

                    void TeleopPeriodic() override;

    .. group-tab:: C++ (Source)

        .. code-block:: cpp

            void Robot::Robot() {
                AddPeriodic([&] {
                    m_motor.Set(m_controller.Calculate(m_encoder.GetRate()));
                }, 10_ms, 10_ms);
            }

            void Robot::TeleopPeriodic() {
                if (m_joystick.GetRawButtonPressed(1)) {
                    if (m_controller.GetSetpoint() == 0.0) {
                        m_controller.SetSetpoint(30.0);
                    } else {
                        m_controller.SetSetpoint(0.0);
                    }
                }
            }

teleopPeriodic() in this example runs every 20ms, and the controller update is run every 10ms with an offset of 10ms from when TeleopPeriodic() runs so that their timeslots don't conflict.
```

Just need someone to turn this into an article and place it into some appropriate section. I think making a "Convenience Features" section under "Advanced Programming" would be great for this and some other miscellaneous articles.

### PR DESCRIPTION
**Pull request by prateekma**
Closes #912

### CODE REVIEW

File: source/docs/software/convenience-features/scheduling-functions.rst [hunk 1]
```diff
@@ -0,0 +1,73 @@
+Scheduling Functions at Custom Frequencies
+==========================================
+In order to be able to run certain functions at a rate that is faster than the default ``TimedRobot`` periodic update rate (20 ms), an ``addPeriodic()`` method has been added to ``TimedRobot``. Previously, teams had to make a ``Notifier`` to run feedback controllers more often than the ``TimedRobot`` loop period of 20 ms (running ``TimedRobot`` more often than this is not advised). Now, users can run feedback controllers more often than the main robot loop, but synchronously with the ``TimedRobot`` periodic functions, eliminating potential thread safety issues.
+
+The ``addPeriodic()`` (Java) / ``AddPeriodic()`` (C++) method takes in a lambda (the function to run), along with the requested frequency and an optional offset from the common starting time. The optional third argument is useful for scheduling a function in a different timeslot relative to the other ``TimedRobot`` periodic methods.
```
Comment (calcmogul) [inline source/docs/software/convenience-features/scheduling-functions.rst:5]:
```suggestion
The ``addPeriodic()`` (Java) / ``AddPeriodic()`` (C++) method takes in a lambda (the function to run), along with the requested period and an optional offset from the common starting time. The optional third argument is useful for scheduling a function in a different timeslot relative to the other ``TimedRobot`` periodic methods.
```

**Review by calcmogul — CHANGES_REQUESTED** [review]

**Comment by Daltz333** [comment]
Should we link to Javadocs?

**Comment by prateekma** [comment]
The Javadocs / Doxygen don't explain anything more than we do here; we've covered everything in them.

File: source/docs/software/convenience-features/scheduling-functions.rst [hunk 2]
```diff
@@ -0,0 +1,73 @@
+Scheduling Functions at Custom Frequencies
+==========================================
+In order to be able to run certain functions at a rate that is faster than the default ``TimedRobot`` periodic update rate (20 ms), an ``addPeriodic()`` method has been added to ``TimedRobot``. Previously, teams had to make a ``Notifier`` to run feedback controllers more often than the ``TimedRobot`` loop period of 20 ms (running ``TimedRobot`` more often than this is not advised). Now, users can run feedback controllers more often than the main robot loop, but synchronously with the ``TimedRobot`` periodic functions, eliminating potential thread safety issues.
```
Comment (calcmogul) [inline source/docs/software/convenience-features/scheduling-functions.rst:3]:
```suggestion
``TimedRobot``'s ``addPeriodic()`` method allows one to run custom methods at a rate faster than the default ``TimedRobot`` periodic update rate (20 ms). Previously, teams had to make a ``Notifier`` to run feedback controllers more often than the ``TimedRobot`` loop period of 20 ms (running ``TimedRobot`` more often than this is not advised). Now, users can run feedback controllers more often than the main robot loop, but synchronously with the ``TimedRobot`` periodic functions, eliminating potential thread safety issues.
```

**Review by calcmogul — CHANGES_REQUESTED** [review]

File: source/docs/software/convenience-features/scheduling-functions.rst (hunk 2 above)
Comment (calcmogul) [reply source/docs/software/convenience-features/scheduling-functions.rst:3]: Just some more wordsmithing.

File: source/docs/software/convenience-features/scheduling-functions.rst [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                @Override
+                public teleopPeriodic() {
+                    if (m_joystick.getRawButtonPressed(1)) {
+                        if (m_controller.getSetpoint() == 0.0) {
+                            m_controller.setSetpoint(30.0);
+                        } else {
+                            m_controller.setSetpoint(0.0);
+                        }
+                    }
+                }
```
Comment (calcmogul) [inline source/docs/software/convenience-features/scheduling-functions.rst:36]: We should add closing braces to the class declarations in the examples.

**Review by calcmogul — APPROVED** [review]

**Review by Daltz333 — APPROVED** [review]

**Pull request merged**